### ISSUE TITLE
[Change Request]: netId in vehicle logs

### ISSUE THREAD

**Issue by MorgVerd** [official-antistasi-community/A3-Antistasi#2798]
### Is the requested change related to a problem?

Vehicle related logs (specifically the garage in this example) do not contain any static identifiers between states.
Links: [fn_removeVehicle](https://github.com/official-antistasi-community/A3-Antistasi/blob/8662513b5cf5504d25bb8a4526116745070efc43/A3A/addons/garage/Core/fn_removeFromPool.sqf#L69), [fn_addVehicle](https://github.com/official-antistasi-community/A3-Antistasi/blob/8662513b5cf5504d25bb8a4526116745070efc43/A3A/addons/garage/Public/fn_addVehicle.sqf#L193). (I'm sure there are more, but this is just about the garage for now).

Both of these logs share a common set of keys:
  - Player Name
  - Player SteamID
  - Vehicle Class
  - Vehicle UID (used by garage system internally).
  
However, between the two log lines mentioned above, the `Vehicle UID` changes, meaning there are no static identifiers to associate loglines with their corresponding vehicle. You could use the steamid + vehicle class but this breaks if the player pulls two of the same vehicle type out.

### Describe the solution you'd like.

**I have very little experience with SQF. This is based on my current/limited understanding from documentation. If I am incorrect please let me know.**

Simply include the vehicle `netId` to vehicle logs. To my knowledge this should remain as a fixed value, and can be used across all events (garage + vehicle events like destroyed etc).

This would allow for logs to be matched against vehicles which is especially useful for external logging etc.

### Additional context

I personally want this change as I have been working on an extended integration between Antistasi and Discord for a private campaign server. The idea for the garage was that the message (below) would be edited as the vehicle state changed (Retrieved from garage, Stored in garage, Destroyed). 

Originally I mistakenly thought the `Vehicle UID` was what I was looking for (a static value corresponding to the vehicle), however as mentioned above this value changes (which I didn't realise until I did my testing after I'd finished making most of the out of server integration code).

![image](https://github.com/official-antistasi-community/A3-Antistasi/assets/25158176/6ea39650-7fdf-45f2-8cbe-8575cf704376)

I then intended to use the antistasi extender, however the garage does not call any events. As per my current understanding this would mean that I'd have to edit the core to add this additional logging, which is unsuitable for the long campaigns we are intending to play (I like to keep my things up to date). After speaking with Bob Murphy he suggested that I should make a feature request issue for it, so I did.

**I will continue attempting to learn SQF. Currently I am not at the level where I would feel comfortable contributing to the project myself. If I manage to figure out how to do it myself I will try to commit it myself but obviously I'd love it if someone with more experience could help out here.**

**Comment by killerswin2** [comment]
If I'm reading this right. You want a id for each vehicle placed in the garage, for each unique vehicle. Okay I can see what you want, but why? Also net ids change, so tracking them is not a good method of tracking. To make this work the way you want you gonna need some significant changes to the garage. I'm gonna defer my judgement to this to @HakonRydland .

**Comment by jaj22** [comment]
Yeah, the netId will change every time a vehicle is "placed" from the garage (because it's actually placing a new vehicle), so it's not that persistent. If you really wanted to track specific vehicles then you'd need to store the vehicle UID on the vehicle when placing, and read it back on storing.

What issue would this be solving?

**Comment by MorgVerd** [comment]
@killerswin2 @jaj22 (pinged for clarification)

Hi, apologies it seems that I haven't explained it as well as I'd hoped (I was on break at work so it was a bit rushed). My intention is simply to track the **vehicle's lifecycle**.  By this I mean the "state" changes it goes through from being spawned to being removed (either by being stored in the garage again, or by being destroyed).

_(When the vehicle is put into then out of the garage it will have a new netId, since its a new "lifecycle")_

This doesn't really solve any "issue", however it allows for the logs to actually reflect what happened to specific vehicles. It also (in my case) allows for the log lines / output to be parsed for external events (discord webhooks).

Ultimately: The log line is being called anyway, adding the network ID just adds a little bit more potential functionality.

**Comment by Tiny-DM** [comment]
Theoretically the vehicle UID system could be changed to be specific per vehicle, with the garage UID saved to the vehicle and read and saved when the vehicle is garaged again, preserving every vehicle being unique and giving every vehicle a true ID. Issue is that a counter would need to run between sessions which may get up to like a thousand over the course of a normal campaign as vehicles are found and destroyed. But that shouldn't be too problematic to save.
